### PR TITLE
Fix autosize bug in variable speed cooling coils.

### LINKED ISSUE #10502: Coil:Cooling:DX:VariableSpeed autosize indicator fails for "Evaporative Condenser Pump Rated Power Consumption"
Issue overview
--------------
The `autosize` IDF entry for "Evaporative Condenser Pump Rated Power Consumption" fails to be recognized ; the number -99999 is read  literally instead.

### Details
Some additional details for this issue (if relevant):
 - Windows
 - v24.2 (discovered in 22.2)

[upload.zip](https://github.com/NREL/EnergyPlus/files/15297648/upload.zip)

### PR DESCRIPTION
**Pull request by tanaya-mankad**
Pull request overview
---------------------
 - Fixes #10502 
 - This PR checks for autosize on the IDF entry "Evaporative Condenser Pump Rated Power Consumption" before checking its value. A new unit test confirms that autosizing is processed.

### Pull Request Author
Add to this list or remove from it as applicable.  This is a simple templated set of guidelines.
 - [x] Title of PR should be user-synopsis style (clearly understandable in a standalone changelog context)
 - [x] Label the PR with at least one of: Defect, Refactoring, NewFeature, Performance, and/or DoNoPublish
 - [x] Pull requests that impact EnergyPlus code must also include unit tests to cover enhancement or defect repair
 - [x] Author should provide a "walkthrough" of relevant code changes using a GitHub code review comment process
 

### Reviewer
This will not be exhaustively relevant to every PR.
 - [ ] Perform a Code Review on GitHub
 - [ ] If branch is behind develop, merge develop and build locally to check for side effects of the merge
 - [ ] If defect, verify by running develop branch and reproducing defect, then running PR and reproducing fix
 - [ ] If feature, test running new feature, try creative ways to break it
 - [ ] CI status: all green or justified
 - [ ] Check that performance is not impacted (CI Linux results include performance check)
 - [ ] Run Unit Test(s) locally
 - [ ] Check any new function arguments for performance impacts
 - [ ] Verify IDF naming conventions and styles, memos and notes and defaults
 - [ ] If new idf included, locally check the err file and other outputs

### CODE REVIEW

File: src/EnergyPlus/VariableSpeedCoils.cc [hunk 1]
```diff
@@ -1033,13 +1033,15 @@ namespace VariableSpeedCoils {
...
             state.dataVariableSpeedCoils->VarSpeedCoil(DXCoilNum).EvapCondPumpElecNomPower = NumArray(10);
 
-            if (state.dataVariableSpeedCoils->VarSpeedCoil(DXCoilNum).EvapCondPumpElecNomPower < 0.0) {
-                ShowSevereError(
-                    state,
-                    format("{}{}=\"{}\", invalid", RoutineName, CurrentModuleObject, state.dataVariableSpeedCoils->VarSpeedCoil(DXCoilNum).Name));
-                ShowContinueError(state, format("...{} cannot be < 0.0.", cNumericFields(10)));
-                ShowContinueError(state, format("...entered value=[{:.2T}].", NumArray(10)));
-                ErrorsFound = true;
+            if (state.dataVariableSpeedCoils->VarSpeedCoil(DXCoilNum).EvapCondPumpElecNomPower != Constant::AutoCalculate) {
```
Comment (rraustad) [inline src/EnergyPlus/VariableSpeedCoils.cc]: The Constant::AutoSize and Constant::AutoCalculate do have the same value but they mean different things. We should probably change one of the values. The auto calculate field is defined with ` \autocalculatable`. So this field, which is autosizable, should compare to the AutoSize variable.

File: tst/EnergyPlus/unit/VariableSpeedCoils.unit.cc [hunk 2]
```diff
@@ -7135,4 +7135,152 @@ TEST_F(EnergyPlusFixture, VariableSpeedCoils_ZeroRatedCoolingCapacity_Test)
...
+                                              CyclingScheme,
+                                              CompressorOp, // compressor on/off. 0 = off; 1= on
+                                              PartLoadFrac,
+                                              SpeedCal,
+                                              SpeedRatio,
+                                              SensLoad,
+                                              LatentLoad,
+                                              OnOffAirFlowRatio);
+}
+
```
Comment (rraustad) [inline tst/EnergyPlus/unit/VariableSpeedCoils.unit.cc]: What is the value of Evaporative Condenser Pump Rated Power Consumption after this call?

File: src/EnergyPlus/VariableSpeedCoils.cc (hunk 1 above)
Comment (tanaya-mankad) [reply src/EnergyPlus/VariableSpeedCoils.cc:1036]: Ah, that's useful to know - the rest of the function is peppered with `Autocalculate`s where "Autosize" is used in the IDF (e.g. Rated Air Flow Rate) - I just copied and pasted. So I imagine those were intended to be `Autosize` too?

Comment (rraustad) [reply src/EnergyPlus/VariableSpeedCoils.cc]: OK, this is correct and the idd shows autosize when it should be autocalculatable. You know an input is auto calculated when there is a rule of thumb used to size the input.

            //     Auto size high speed evap condenser pump power to Total Capacity * 0.004266 w/w (15 w/ton)
            EvapCondPumpElecNomPowerDes = varSpeedCoil.RatedCapCoolTotal * 0.004266;

**Comment by rraustad** [comment]
@tanaya-mankad are there other changes you expect to make? These changes look fine for the issue.

**Comment by tanaya-mankad** [comment]
Yes, this is the (greatly simplified) set of changes. Thanks.

**Review by rraustad — APPROVED** [review]
These changes look good to me.

**Comment by Myoldmopar** [comment]
@tanaya-mankad, this has a conflict in the unit test code.  It's otherwise ready, so if you could resolve that, I'll give it a quick once over and get it merged in.  My guess is that two different unit tests were added to the same spot in the unit test file.  It's either that, or one of the namespace-ish refactors just changed a line.  If you hit trouble, let me know and I can give it a shot.

**Comment by tanaya-mankad** [comment]
Hi @Myoldmopar , I'm not finding the failure in the Dashboard. Could you point me to a link?

**Comment by Myoldmopar** [comment]
It's a merge conflict here on the pr, I'm not sure I can send a link to it 🙃

**Comment by tanaya-mankad** [comment]
@Myoldmopar I misunderstood the error, but I found it when I merged develop just now (I was on holiday). Hopefully the CI will be happy soon!

**Comment by dareumnam** [comment]
The conflict in the unit test code has been resolved, and the CI is all green. This PR is ready to be merged. @Myoldmopar

**Comment by Myoldmopar** [comment]
Yep everything seems to look good here and locally.  Thanks all, merging.

**Pull request merged**